src/paragraphcommand.js

```diff
@@ -0,0 +1,79 @@
+/**
+ * @license Copyright (c) 2003-2017, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+/**
+ * @module paragraph/paragraphcommand
+ */
+
+import Command from '@ckeditor/ckeditor5-core/src/command/command';
+
+/**
+ * The paragraph command.
+ *
+ * @extends module:core/command/command~Command
+ */
+export default class ParagraphCommand extends Command {
+	/**
+	 * Creates an instance of the command.
+	 *
+	 * @param {module:core/editor/editor~Editor} editor Editor instance.
+	 */
+	constructor( editor ) {
+		super( editor );
+
+		/**
+		 * Value of the command, indicating whether it is applied in the context
+		 * of current {@link module:engine/model/document~Document#selection selection}.
+		 *
+		 * @readonly
+		 * @observable
+		 * @member {Boolean}
+		 */
+		this.set( 'value', false );
+
+		// Update current value each time changes are done on document.
+		this.listenTo( editor.document, 'changesDone', () => this._updateValue() );
+	}
+
+	/**
+	 * Executes command.
+	 *
+	 * @protected
+	 * @param {Object} [options] Options for executed command.
+	 * @param {module:engine/model/batch~Batch} [options.batch] Batch to collect all the change steps.
+	 * New batch will be created if this option is not set.
+	 * @param {module:engine/model/selection~Selection} [options.selection] Selection the command should be applied to.
+	 * By default, if not provided, the command is applied to {@link module:engine/model/document~Document#selection}.
+	 */
+	_doExecute( options = {} ) {
+		if ( this.value && !options.selection ) {
```

Comment:
The value of the command is determined by `this.editor.document.selection`. So if it is true, there's no point in applying it once again, unless there's a custom selection to work on.

But my fault is somewhere else. The `#value` is determined by `editor.document.selection.getSelectedBlocks().next().value`. So it says whether the start of the selection is anchored in paragraph or not. The command, OTOH, might apply to a wider selection, containing more than a single paragraph, but anchored in it nevertheless.

So `value === true` does not mean there's no need to apply to the command. Gonna fix it.